packages/odyssey-react/src/utils/withStyles/withStyles.tsx

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { forwardRef, useLayoutEffect } from "react";
+import hoistNonReactStatics from "hoist-non-react-statics";
+import OStyleSheet from "./stylesheet";
+import type { ComponentType, ElementRef } from "react";
+
+export type Template = () => string;
+type SourceStyles = Record<string, string>;
+interface TranspiledStyles {
+  __digest: string;
+  __template: Template;
+}
+type Styles = SourceStyles | TranspiledStyles;
+
+const cache = new Map<string, OStyleSheet>();
```

Comment:
in the future I'd like this cache to be browser frame wide to support microfrontends or composed UIs

Reply:
Does it have to be a map? Set will not work?